HelpSource/Classes/DigitalOut.schelp

```diff
@@ -0,0 +1,71 @@
+class:: DigitalOut
+summary:: Write data to a digital input of the BELA board
+related:: Classes/AnalogIn, Classes/AnalogOut, Classes/DigitalIn, Classes/DigitalIO
+categories::  UGens>BELA
+
+
+Description::
+
+Write digital data to a digital output of the BELA board.
+
+note::
+This UGen only works on BeLa.
+::
+
+note::
+If you want to modulate the pin number, you should use the UGen link::Classes/DigitalIO::
+::
+
+classmethods::
+
+method::ar
+
+argument::digitalPin
+
+Digital pin number to write to. Pin numbers begin at 0. This value cannot be modulated.
+
+argument::output
+
+Value to write out to the pin - the value will be 1 when the argument is larger than 0, otherwise 0.
+
+argument::writeMode
+
+Mode of writing to the output, this can be 0 (using Bela's DigitalWrite and only when value changes) or 1 (using Bela's DigitalWriteOnce). This value cannot be modulated.
```

Comment:
> Mode of writing to the output, this can be 0 (using Bela's DigitalWrite and only when value changes) or 1 (using Bela's DigitalWriteOnce). This value cannot be modulated.

This is a mode that is quite hardware specific, and reflects how this is implemented in the Bela backend, and reflects a certain efficiency on the hardware level. Either the value is written to the hardware at each calculation step of the UGen (mode 1) (at audio rate), or only when the value changes (mode 0).

So here goes for an alternative description:

```
argument::writeMode

Mode of writing to the output:
0: only write the value to the hardware when the value changes
1: write the value each time the output of the UGen is calculated, i.e. at each audio sample.
```

With at the bottom of the helpfile:

```
Bela specific notes

The write mode 0 is implemented using the C++ function digitalWrite of the Bela framework.
The write mode 1 is implemented using the C++ function digitalWriteOnce of the Bela framework.
```

(my schelp formatting is a bit rusty). I also don't have a current checkout of this supercollider handy to make a commit myself... so I hope my comment helps for someone who does!

Reply:
In theory, mode 0 would be more CPU-efficient for occasional changes, while mode 1 would be more efficient for frequent changes. I think back in 2016 there may have been a performance difference between the two, however the Bela implementation has improved since and I do not measure a CPU difference between the two now, so I think this is a good time to get rid of the two modes.